library/public/engine.h

```diff
@@ -47,22 +48,91 @@ class F3D_EXPORT engine
 {
 public:
   /**
-   * Engine constructor, choose the window type using the enum.
-   * see window.h for details about the window.
-   * When using window::Type::NONE, window and interactor will not be provided by the engine.
-   * When using window::Type::EXTERNAL, interactor will not be provided by the engine.
-   * All objects instances will be created on construction.
-   * Default is window::Type::NATIVE.
-   * Throw a no_window_exception when using a Using window::Type::EXTERNAL without the right cmake
-   * option.
+   * Create an engine with an automatic window.
+   * Linux: Try GLX, then EGL, then OSMesa
+   * Windows: Try Win32, then EGL
+   * macOS: Always use Cocoa
+   * Optionally, the window can be hidden by setting offscreen to true.
    */
-  explicit engine(window::Type windowType = window::Type::NATIVE);
+  static engine create(bool offscreen = false);
+
+  /**
+   * Create an engine with no window.
+   */
+  static engine createNone();
+
+  /**
+   * Create an engine with a GLX window.
+   * Works on Linux only.
+   * Optionally, the window can be hidden by setting offscreen to true.
+   */
+  static engine createGLX(bool offscreen = false);
+
+  /**
+   * Create an engine with a WGL window.
+   * Works on Windows only.
+   * Optionally, the window can be hidden by setting offscreen to true.
+   */
+  static engine createWGL(bool offscreen = false);
+
+  /**
+   * Create an engine with an EGL window.
+   * If several GPU are available, the environment variable
+   * `VTK_DEFAULT_EGL_DEVICE_INDEX` allows its selection.
+   * Optionally, the window can be hidden by setting offscreen to true.
+   */
+  static engine createEGL(bool offscreen = false);
+
+  /**
+   * Create an engine with an OSMesa window.
+   */
+  static engine createOSMesa();
+
+  /**
+   * Create an engine with an external window.
+   * A context to retrieve OpenGL symbols is required.
+   * The context can be nullptr for an external WGL or Cocoa window.
```

Comment:
can it ? crazy.

Reply:
not true for WGL, I need to check with macOS